runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -149,6 +150,38 @@ private void txBeginInternal(boolean allowNestedTransactions) {
      */
     private <K extends Message, V extends Message, M extends Message>
     void validateWrite(@Nonnull Table<K, V, M> table, K key, boolean validateKey) {
+        baseValidateWrite(table, key, validateKey);
+
+        txnType |= WRITE_ONLY;
+        tablesInTxn.putIfAbsent(table.getStreamUUID(), table);
+    }
+
+    /**
+     * All write api must be validate to ensure that the table belongs to the namespace.
+     *
+     * @param table       - table being written to
+     * @param key         - key used in the transaction to check for null.
+     * @param validateKey - should key be validated for null.
+     * @param <K>         - type of the key
+     * @param <V>         - type of the payload/value
+     * @param <M>         - type of the metadata
+     */
+    private <K extends Message, V extends Message, M extends Message>
+    void validateWrite(@Nonnull Table<K, V, M> table, K key, V value, M metadata, boolean validateKey) {
+        baseValidateWrite(table, key, validateKey);
+        if (table.getMetadataClass() == null && metadata != null) {
+            throw new IllegalArgumentException("Metadata schema for table " + table.getFullyQualifiedTableName() + " is defined as NULL, non-null metadata is not allowed.");
+        }
+        if (table.getValueClass() == null && value != null) {
```

Comment:
I don't think we support null payloads, perhaps this check can be moved into openTable itself?

Reply:
It should cause currently nothing prevents value or key being null. The annotation in place is only informational.

Reply:
I've added the check on openTable